### ISSUE TITLE
New-NavContainer works, but cannot login in Chrome

### ISSUE THREAD

**Issue by kevindstanley1988** [microsoft/navcontainerhelper#886]
I have been using chrome on windows server 2019 to access BC dockers for months now. I went to login today and the I key in username/password and click sign in and the form clears and nothing happens. 

I thought maybe it was my container, so I created a fresh container and got the same behavior. 

I have very basic parameters for my docker setup. 
`
New-NavContainer $containerName -accept_eula -accept_outdated -imageName "mcr.microsoft.com/businesscentral/onprem:na" -licenseFile "C:\license.flf" -shortcuts Desktop -updateHosts -auth NavUserPassword -memoryLimit "4G"
`

The symbols will pull, so I know the container is working.
![image](https://user-images.githubusercontent.com/5685997/75887716-34f58680-5df8-11ea-8fa5-10f20bdd5ddb.png)

**Comment by freddydk** [comment]
Could you provide the output of the New-NavContainer command?
and did you try with other browsers and inprivate mode?
and what is the error?

**Comment by kevindstanley1988** [comment]
I also tried in private mode. There is no error, it just clears the form in chrome. I created different named containers too so in case it was something to do with the hostname, still same issue though.

powershell output
```
cmdlet New-Container.ps1 at command pipeline position 1
Supply values for the following parameters:
containerName: cronus
Windows PowerShell credential request.
Using NavUserPassword Authentication. Please enter username/password for the Containter.
User: kstanley
Password for user kstanley: ********
NavContainerHelper is version 0.6.4.19
NavContainerHelper is running as administrator
Host is Microsoft Windows Server 2019 Datacenter - ltsc2019
Docker Client Version is 19.03.2
Docker Server Version is 19.03.2
Using image mcr.microsoft.com/businesscentral/onprem:na-ltsc2019
Removing container cronus
Removing cronus from host hosts file
Removing C:\ProgramData\NavContainerHelper\Extensions\cronus
Creating Container cronus
Version: 15.1.37881.38071-na
Style: onprem
Platform: 15.0.37898.0
Generic Tag: 0.0.9.96
Container OS Version: 10.0.17763.805 (ltsc2019)
Host OS Version: 10.0.17763.678 (ltsc2019)
WARNING: The container operating system matches the host operating system, but the revision is different.
If you encounter issues, you might want to specify -isolation hyperv
Using locale en-US
Using process isolation
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Using license file C:\ProgramData\NavContainerHelper\license\20200109 BC License.flf
Files in C:\ProgramData\NavContainerHelper\Extensions\cronus\my:
- AdditionalOutput.ps1
- license.flf
- MainLoop.ps1
- SetupVariables.ps1
- SetupWebClient.ps1
- updatehosts.ps1
Creating container cronus from image mcr.microsoft.com/businesscentral/onprem:na-ltsc2019
3f7187debd93d668e06258c21db994ce2b8e6a1bf18cb524aa6ab806adc5c332
Waiting for container cronus to be ready
Initializing...
Setting host.docker.internal to 192.168.1.62 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to 192.168.1.62 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.30.80.1 in container hosts file
Starting Container
Hostname is
PublicDnsName is cronus
Using NavUserPassword Authentication
Starting Local SQL Server
Starting Internet Information Server
Creating Self Signed Certificate
Self Signed Certificate Thumbprint 1DDF14F9063D1E936E238E78930B1CFDE834DDB1
Modifying Service Tier Config File with Instance Specific Settings
Starting Service Tier
Registering event sources
Creating DotNetCore Web Server Instance
Using license file 'c:\run\my\license.flf'
Import License
Creating http download site
Setting SA Password and enabling SA
Creating kstanley as SQL User and add to sysadmin
Creating SUPER user
Container IP Address: 172.30.94.81
Container Hostname  :
Container Dns Name  : cronus
Web Client          : http://cronus/BC/
Dev. Server         : http://cronus
Dev. ServerInstance : BC
Setting cronus to 172.30.94.81 in host hosts file

Files:
http://cronus:8080/al-4.0.194000.vsix

WARNING: You are running a container which is 122 days old.
Microsoft recommends that you always run the latest version of our containers.

Initialization took 63 seconds
Ready for connections!
Reading CustomSettings.config from cronus
Creating Desktop Shortcuts for cronus
Creating .net Assembly Reference Folder for VS Code
Copying DLLs from C:\Windows\assembly to assemblyProbingPath
Copying DLLs from C:\Program Files\Microsoft Dynamics NAV\150\Service to assemblyProbingPath
Copying DLLs from C:\Program Files (x86)\Microsoft Dynamics NAV\150\RoleTailored Client to assemblyProbingPath
Copying DLLs from C:\Test Assemblies\Mock Assemblies to assemblyProbingPath
Copying DLLs from C:\Program Files (x86)\Open XML SDK to assemblyProbingPath
```

Here is the network tab in chrome, I can see that the signin is being sent, but it then loads back to the sign in url http://masterupdate/BC/SignIn?ReturnUrl=%2FBC%2F. This is a freshly loaded network tab, I key in my user and my password and it returns just a cleared form. I triple checked my password too, so it isn't that - and if I intentionally type the wrong password I actually get the incorect user/password error. 
![image](https://user-images.githubusercontent.com/5685997/75901704-f0281a80-5e0c-11ea-9209-6d9590bd3d3d.png)

**Comment by freddydk** [comment]
have you tried -isolation hyperv?

**Comment by kevindstanley1988** [comment]
I haven't installed hyper-v on this server. This server has more than a few docker containers. We have been using them actively for almost a year, since spring release of last year and then upgraded to fall release. The containers created weeks ago and have been working all stopped being able to login with chrome this morning.

I am finding that if I open the client in firefox it is working fine. I have uninstalled and reinstalled chrome to no luck - cleared cache/used incognito mode/etc.. I am at a loss on what to do here.

**Comment by freddydk** [comment]
If the client works with FireFox and other browsers - it doesn't seem to be a problem with the container. I haven't seen this before.
If you try to spin up an old image (like 1904-cu1-us-ltsc2019), does that work with Chrome?

**Comment by freddydk** [comment]
In Chrome, if you click in the address bar and press Ctrl+C - then paste the URL - does it say http or https?
On my machine - it says:
https://dev/BC/

which won't work because the container is not created using ssl.
You could also try

-useSSL

in the command line to see whether it works with SSL.

**Comment by freddydk** [comment]
I can repro this using the container name dev. Somehow Chrome thinks that it needs to use https for dev. I also cannot download the .cer file using http and Chrome complains if I create the container using -useSSL.
If I name the container devxxx - then it works.
I don't get a blank page though  - I get an error.

**Comment by NAVspecialist** [comment]
Hello guys
Try this:
In the browsermenu on chrome. type 

chrome://net-internals 

Find the field option Delete domain security policies (at the bottom) 
Enter the domain name and press the delete-button.

It was so annoying that I decided no longer to use dev as my container name.

I have been struggling with the same. Google have bought the toplevel domain .dev and that might be the issue.

Regards
Palle

**Comment by kevindstanley1988** [comment]
> I can repro this using the container name dev. Somehow Chrome thinks that it needs to use https for dev. I also cannot download the .cer file using http and Chrome complains if I create the container using -useSSL.
> If I name the container devxxx - then it works.
> I don't get a blank page though - I get an error.

No matter what they are named are failing to login(clearing the login form). I just tried to create one with -useSSL and I had to manually install the cert file in windows certs for chrome to pick it up. I could not download the self signed cert through chromes interface. Using -useSSL and installing the cert manually did let me get past the login screen and get to BC, but this is a whole lot of manual steps to make a container that was working a few days ago start working again.

**Comment by freddydk** [comment]
Well, I hope you know that this isn't something we changed, we didn't change anything.

Somehow Chrome on your machine caused Chrome to demand SSL for your containers. Whether it is a Chrome update or a policy change, I don't know, but I will investigate when I get back from vacation.

In the meantime, you would need to use a different browser or use SSL. I am pretty sure the manual steps above can be automated - which is what I would try to do, if I discover that this is the only way to go.

**Comment by freddydk** [comment]
I am trying to find out whether this only is a https problem.

If you add these lines to the script after creating the container:

```
$containerName = "dev"
Copy-FileFromNavContainer -containerName $containerName -containerPath "c:\run\certificate.cer" -localPath "C:\ProgramData\NavContainerHelper\Extensions\$containerName\certificate.cer"
Import-Certificate -FilePath "C:\ProgramData\NavContainerHelper\Extensions\$containerName\certificate.cer" -CertStoreLocation "cert:\localMachine\Root"
```

Does it then work as expected?

**Comment by kevindstanley1988** [comment]
> I am trying to find out whether this only is a https problem.
> 
> If you add these lines to the script after creating the container:
> 
> ```
> $containerName = "dev"
> Copy-FileFromNavContainer -containerName $containerName -containerPath "c:\run\certificate.cer" -localPath "C:\ProgramData\NavContainerHelper\Extensions\$containerName\certificate.cer"
> Import-Certificate -FilePath "C:\ProgramData\NavContainerHelper\Extensions\$containerName\certificate.cer" -CertStoreLocation "cert:\localMachine\Root"
> ```
> 
> Does it then work as expected?

This works exactly as expected(loads after BC after I login through https://xxxx/BC), this should work fine for our use cases, it is an automated approach to what I was doing manually for testing https, but thank your for writing it up. 

I am still very unsure what could have caused chrome to stop accepting standard http for these docker containers, but as long as I have a workaround I guess this is fine. It seems odd that the container name 'dev' stopped working globally for everyone at the same time this happened for our different named containers, who knows maybe it was part of a chrome update and the next one will remedy it.

**Comment by freddydk** [comment]
I am running Google Chrome Version 80.0.3987.132
the name dev stopped working way back, I can access other container names.
If might be a policy on your machine or some other setting.

I might add a parameter to new-navcontainer, like -installCertOnHost, which will (if using self signed cert) install the cert on the host and remove it upon removal of the container.

**Comment by freddydk** [comment]
Next version of containerhelper implements a parameter called InstallcertificateOnHost, which adds a few output: 
![image](https://user-images.githubusercontent.com/10775043/76314496-37178380-62d7-11ea-8dcb-8efb15bc8c0e.png)

**Comment by freddydk** [comment]
ships in 0.6.5.1 - in a few hours.

**Comment by freddydk** [comment]
Fixed in 0.6.5.1 and 0.6.5.2
